governance/default_test.go

```diff
@@ -537,3 +537,24 @@ func TestCypressGenesisHash(t *testing.T) {
 		t.Errorf("Generated hash is not equal to Cypress's hash. Want %v, Have %v", cypressHash.String(), block.Hash().String())
 	}
 }
+
+func TestWriteGovernance_idxCache(t *testing.T) {
+	gov := getGovernance()
+
+	tstMap := copyMap(testGovernanceMap)
+
+	src := NewGovernanceSet()
+	delta := NewGovernanceSet()
+	src.Import(tstMap)
+
+	gov.WriteGovernance(30, src, delta)
+	gov.WriteGovernance(30, src, delta)
+	gov.WriteGovernance(60, src, delta)
+	gov.WriteGovernance(60, src, delta)
+	gov.WriteGovernance(50, src, delta)
+
+	// idxCache should have 0, 30 and 60
+	if len(gov.idxCache) != 3 && gov.idxCache[len(gov.idxCache)-1] != 60 {
```

Comment:
```suggestion
	if len(gov.idxCache) != 3 || gov.idxCache[len(gov.idxCache)-1] != 60 {
```
`&&` needs to be replaced by `||`, doesn't it?